### PR TITLE
Replace get/setSex with get/setGender

### PR DESCRIPTION
**Pull request by heiglandreas**
As explained in #94 it's a good idea to use gender instead of sex. 

To not break BC the get/setSex are still available but are marked as deprecated. The current functionality works as before. In addition to that it is possible to set `other` as gender and the default now is to return `unknown` instead of NULL.

### CODE REVIEW

File: src/Common/Entity/User.php [hunk 1]
```diff
@@ -47,11 +49,11 @@ class User extends \stdClass
...
 
     /**
-     * Should be female or male
+     * One of the GENDER_-constants
      *
-     * @var string|null
+     * @var string
      */
-    protected $sex;
+    protected $gender = GENDER_UNKNOWN;
```
Comment (ADmad) [inline src/Common/Entity/User.php]: Shouldn't it be `self::GENDER_UNKNOWN`?

File: src/Common/Entity/User.php [hunk 2]
```diff
@@ -80,22 +82,43 @@ public function setBirthday(?\DateTime $birthday): void
...
     /**
      * @param string $sex
      */
-    public function setSex(string $sex): void
+    public function setGender(string $gender): void
     {
-        if ($sex !== self::SEX_MALE && $sex !== self::SEX_FEMALE) {
-            throw new \InvalidArgumentException('Argument $sex is not valid');
+        $genders = [
+            GENDER_OTHER,
```
Comment (ADmad) [inline src/Common/Entity/User.php]: All constants should have `static::` prefix.

File: src/Common/Entity/User.php [hunk 3]
```diff
@@ -80,22 +82,43 @@ public function setBirthday(?\DateTime $birthday): void
...
+            GENDER_MALE,
+            GENDER_FEMALE,
+        ];
+        if (! $gender in_array($genders)) {
```
Comment (ADmad) [inline src/Common/Entity/User.php]: This isn't value code :)

File: src/Common/Entity/User.php (hunk 1 above)
Comment (heiglandreas) [reply src/Common/Entity/User.php]: You got  me there! Sure!

**Comment by heiglandreas** [comment]
The removal of the protected proprety `$sex` as well as modifying the return-type of the method `getSex` from `?string` to `string` is technically a BC break. 

As the method now does no longer return `null` any checks expecting `null` can still be executed on the returned value, so it's not really going to break code.

And the removal of the `$sex` property only is a BC break for classes that extend the User-class. Currently that should not have happened. For the future I would advise to either mark the class as `final` or set the visibility of the properties to `private` instead of `public` or `protected` 

See result of `docker run --rm -v `pwd`:/app nyholm/roave-bc-check

```
[BC] REMOVED: Property SocialConnect\Common\Entity\User#$sex was removed
[BC] CHANGED: The return type of SocialConnect\Common\Entity\User#getSex() changed from ?string to string
[BC] REMOVED: Property SocialConnect\Common\Entity\User#$sex was removed
[BC] CHANGED: The return type of SocialConnect\Common\Entity\User#getSex() changed from ?string to string
4 backwards-incompatible changes detected
```

**Pull request closed without merging**